### PR TITLE
resnet amp optimization

### PR DESCRIPTION
**Pull request by Jie-Fang**
Change steps:

1. fix bugs on python3

### CODE REVIEW

**Comment by Jie-Fang** [comment]
2. Add NHWC formart option for resnet

File: benchmark/collective/resnet/utils/reader_cv2.py [hunk 1]
```diff
@@ -61,13 +61,13 @@ def reader():
             for line in lines:
                 if mode == 'train':
                     img_path, label = line.split()
-                    img_path = img_path.replace("JPEG", "jpeg")
-                    img_path = os.path.join(data_dir, "train", img_path)
+                    #img_path = img_path.replace("JPEG", "jpeg")
```
Comment (gongweibao) [inline benchmark/collective/resnet/utils/reader_cv2.py]: This need not to be changed.

File: benchmark/collective/resnet/utils/reader_cv2.py [hunk 2]
```diff
@@ -19,10 +19,10 @@
 import paddle.fluid as fluid
 import random
 
-from img_tool import process_image
+from .img_tool import process_image
 DATA_DIR=""
 
-THREAD=4
+THREAD=8
```
Comment (gongweibao) [inline benchmark/collective/resnet/utils/reader_cv2.py]: Too many threads for fleet now!

File: benchmark/collective/resnet/utils/img_tool.py [hunk 3]
```diff
@@ -106,7 +106,7 @@ def process_image(sample, mode, color_jitter, rotate, settings,
     raw_img = sample[0]
         
     #img_data = np.fromstring(raw_img, dtype='uint8')
-    img_data = np.frombuffer(raw_img, dtype='uint8')
+    #img_data = np.frombuffer(raw_img, dtype='uint8')
```
Comment (gongweibao) [inline benchmark/collective/resnet/utils/img_tool.py]: Cleanup these three lines?

Comment (Jie-Fang) [reply benchmark/collective/resnet/utils/img_tool.py]: Done

File: benchmark/collective/resnet/utils/reader_cv2.py (hunk 2 above)
Comment (Jie-Fang) [reply benchmark/collective/resnet/utils/reader_cv2.py]: Done

**Pull request merged**